.test-infra/jenkins/job_beam_Java_UnitTest.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import common_job_properties
+
+// This is the Java Jenkins job which runs the current set of standard unit tests.
+mavenJob('beam_Java_UnitTest') {
+  description('Runs Java Surefire unit tests. Designed to be run by a pipeline job.')
+
+  // Set standard properties for a job which is part of a pipeline.
+  common_job_properties.setPipelineJobProperties(delegate, 20, "Java Unit Tests")
+  // Set standard properties for a job which pulls artifacts from an upstream job.
+  common_job_properties.setPipelineDownstreamJobProperties(delegate, 'beam_Java_Build')
+
+  // Construct Maven goals for this job.
+  profiles = [
+    'direct-runner',
+    'dataflow-runner',
+    'spark-runner',
+    'flink-runner',
+    'apex-runner'
+  ]
+  args = [
+    '-B',
+    '-e',
+    "-P${profiles.join(',')}",
+    'surefire:test@default-test',
+    'coveralls:report', // TODO: Will this work? Can't verify on my own Jenkins due to no coveralls.
```

Comment:
<!--new_thread; commit:13434ff7764e80f3ff96f6120b9100f537df0f36; resolved:0-->
Is it possible to run collections of targets like this within a profile, or not worth the maven-side complexity?

Reply:
My impression was that it was not worth the maven-side complexity; if this proves to be a maintenance burden I'm happy to put it into a profile instead, but it's easier to ship it here first and this PR is already fairly complex so I figured just making groovy changes first would make review and understanding what happened easier.